lib/Dialect/FIRRTL/Transforms/EmitOMIR.cpp

```diff
@@ -353,7 +687,11 @@ void EmitOMIRPass::makeTrackerAbsolute(Tracker &tracker) {
   auto builder = OpBuilder::atBlockBegin(getOperation().getBody());
 
   // Pick a name for the NLA that doesn't collide with anything.
-  auto opName = tracker.op->getAttrOfType<StringAttr>("name");
+  StringAttr opName;
+  if (auto module = dyn_cast<FModuleLike>(tracker.op))
+    opName = module.moduleNameAttr();
+  else
+    opName = tracker.op->getAttrOfType<StringAttr>("name");
   auto nlaName = circuitNamespace->newName("omir_nla_" + opName.getValue());
```

Comment:
This change, and the other two below should be scrutinized. These are the only non-mechanical changes in this patch.

Reply:
I tested this patch on two internal designs and the output OMIR was the same in both cases. 🎊

Reply:
This looks okay to me, especially since we're running this through `circuitNamespace` so the result should be unique. And I guess the actual name itself is not all that critical.